src/cluster_legacy.c

```diff
@@ -2943,6 +3059,21 @@ int clusterProcessPacket(clusterLink *link) {
      * because of Pub/Sub. */
     if (sender) sender->data_received = now;
 
+    if (sender && (type == CLUSTERMSG_TYPE_PUBLISH_LIGHT || type == CLUSTERMSG_TYPE_PUBLISHSHARD_LIGHT) &&
+        nodeSupportsLightMsgHdr(sender)) {
+        return pubsubProcessLightPacket(link, type);
```

Comment:
There's an address sanitizer error for this function, that `hdr->flags` is accessed for a light message.

We have to reorder things in this function. We can't access fields like `hdr->flags` or `hdr->mflags` for light messages.

Suggested order of things:

1. Stuff like sender->data_received = now;
2. If type == light, process light and return.
3. Stuff like `flags = ntohs(hdr->flags)` (needs a non-light message).

Reply:
Yeah. I didnt look at those yet, but makes sense